modules/core/src/viewports/viewport.js

```diff
@@ -445,11 +445,13 @@ export default class Viewport {
      */
 
     // matrix for conversion from world location to screen (pixel) coordinates
-    const m = createMat4();
-    mat4_scale(m, m, [this.width / 2, -this.height / 2, 1]);
-    mat4_translate(m, m, [1, -1, 0]);
-    mat4_multiply(m, m, this.viewProjectionMatrix);
-    this.pixelProjectionMatrix = m;
+    const viewportMatrix = createMat4(); // matrix from NDC to viewport.
```

Comment:
No function changes here except, adding a new matrix `this.viewportMatrix`, that can convert from NDC to viewport (screen) space.